packages/headless/src/controllers/case-input/headless-case-input.ts

```diff
@@ -0,0 +1,152 @@
+import {updateCaseInput} from '../../features/case-input/case-input-actions';
+import {fetchCaseClassifications} from '../../features/case-field/case-field-actions';
+import {logUpdateCaseField} from '../../features/case-assist/case-assist-analytics-actions';
+import {CaseAssistEngine} from '../../app/case-assist-engine/case-assist-engine';
+import {
+  CaseAssistConfigurationSection,
+  CaseFieldSection,
+  CaseInputSection,
+  ConfigurationSection,
+  DocumentSuggestionSection,
+} from '../../state/state-sections';
+import {buildController, Controller} from '../controller/headless-controller';
+import {
+  caseAssistConfiguration,
+  caseField,
+  caseInput,
+  configuration,
+  documentSuggestion,
+} from '../../app/reducers';
+import {Schema, StringValue} from '@coveo/bueno';
+import {validateOptions} from '../../utils/validate-payload';
+import {loadReducerError} from '../../utils/errors';
+import {fetchDocumentSuggestions} from '../../features/document-suggestion/document-suggestion-actions';
+
+export interface CaseInputOptions {
+  field: string;
+}
+
+export interface CaseInputProps {
+  options: CaseInputOptions;
+}
+
+function validateCaseInputOptions(
+  engine: CaseAssistEngine,
+  options: Partial<CaseInputOptions> | undefined
+) {
+  const schema = new Schema<CaseInputOptions>({
+    field: new StringValue({
+      required: true,
+      emptyAllowed: false,
+    }),
```

Comment:
You can use the `requiredNonEmptyString` shorthand.

```suggestion
    field: requiredNonEmptyString,
```